### PR TITLE
Firestore user repo

### PR DESCRIPTION
**Pull request by yatharthranjan**

### CODE REVIEW

File: src/main/kotlin/org/radarbase/push/integration/garmin/user/firebase/CovidCollabFirestore.kt [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        val userDocumentSnapshot = getDocument(uuid, userCollection)
+        return createUser(userDocumentSnapshot, garminDocumentSnapshot)
+    }
+
+    protected fun createUser(
+        userSnapshot: DocumentSnapshot, garminSnapshot: DocumentSnapshot
+    ): FirebaseUser? {
+        if (!garminSnapshot.contains(OAUTH_KEY)) {
+            logger.warn(
+                "The ${OAUTH_KEY} key for user {} in the fitbit" +
```
Comment (callumstew) [inline src/main/kotlin/org/radarbase/push/integration/garmin/user/firebase/CovidCollabFirestore.kt:67]: Change log message to garmin

**Review by callumstew — COMMENTED** [review]
Looks good to me, although obviously I'm no expert. There's one log message which still references fitbit, but functionally looks good.

File: .env [hunk 2]
```diff
@@ -1 +1,2 @@
 KAFKA_CONFLUENT_VERSION=6.1.0
+GOOGLE_CREDENTIALS_FILE_PATH=/Users/yatharth/Downloads/mind-course-firebase-adminsdk-8nyar-9d05087b3d.json
```
Comment (afolarin) [inline .env:2]: do you want the google cred file path hardcoded?

**Review by afolarin — APPROVED** [review]
LGTM

File: .env (hunk 2 above)
Comment (yatharthranjan) [reply .env:2]: Thanks will remove.

File: src/main/kotlin/org/radarbase/push/integration/garmin/user/firebase/CovidCollabFirestore.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    @Synchronized
+    private fun updateUser(garminDocumentSnapshot: DocumentSnapshot) {
+        try {
+            val user: FirebaseUser? = createUser(
+                getDocument(garminDocumentSnapshot.id, userCollection), garminDocumentSnapshot
+            )
+            logger.debug("User to be updated: {}", user)
+            if (checkValidUser(user)) {
+                val user1: FirebaseUser? = cachedUsers.put(user?.id ?: return, user)
+                if (user1 == null) {
```
Comment (mpgxvii) [inline src/main/kotlin/org/radarbase/push/integration/garmin/user/firebase/CovidCollabFirestore.kt:109]: Would it be better to rename `user1` to something else (maybe `existingUser`)?

**Review by mpgxvii — APPROVED** [review]
LGTM just a minor question/comment.

File: src/main/kotlin/org/radarbase/push/integration/garmin/user/firebase/CovidCollabFirestore.kt (hunk 3 above)
Comment (yatharthranjan) [reply src/main/kotlin/org/radarbase/push/integration/garmin/user/firebase/CovidCollabFirestore.kt]: Yes sure, updated now with some other improvements.

**Pull request merged**